gemrb/core/GUI/GUIAnimation.h

```diff
@@ -144,7 +144,6 @@ class GEM_EXPORT SpriteAnimation : public GUIAnimation<Holder<Sprite2D>> {
 	explicit SpriteAnimation(std::shared_ptr<const AnimationFactory> af, int Cycle = 0);
 	//report if the current resource is the same as descripted by the params
 	void SetPaletteGradients(const ieDword *col);
-	bool SameResource(const SpriteAnimation*) const;
```

Comment:
This whole thing was added for the pause button, so that's a good test case. In bg2 the end visual requires 3 animations, but as long as we can set flags on them, I don't think we need anything like SameResource.

Reply:
good to know. indeed this breaks that, so some further changes are needed. Im still unsure of how this was uses, since all it would do is just delete the incoming animation and nothing else. Seems like just don't set animations you don't want should fix it, but 🤷🏻‍♂️

Reply:
I suppose the critical part must have been the `if (anim->cycle != c) return false;`. I think on this, but worst case is this check just moves to `Button`.

Reply:
ok, this _does_ actually appear to still work now. Seems the previous breakage I observed was due to file path formation issues. We only call `Button.SetAnimation` in `CreateClockButton` which only runs once so I still don't understand the origin of `SameResource`, but it seems to be history at this point.